### PR TITLE
Fix code

### PR DESCRIPTION
**Pull request by omargalvan18**

### CODE REVIEW

**Review by RajeshC03 — COMMENTED** [review]
Good addition of fclose(file); to properly close the file. However, consider adding a NULL check to prevent potential segmentation faults if file was not opened successfully

**Comment by Sandeep-pan** [comment]
Missing Unit Tests for Negative Cases
The test currently only checks for a valid case. It would be useful to add tests for:

Incorrect magic number (like header[0] = 'X')

Unsupported version (like version = 999)

Short/truncated header (like sizeof(struct sc_seccomp_file_header))

That way, we make sure the function correctly fails when expected.

Memory and File Handling
In the test:

The FILE * returned from the validation function isn't closed — should add a fclose(file); at the end.

The profile path from g_file_open_tmp() is not freed — could add g_free(profile); as cleanup.

Error Feedback
Right now, if validation fails, we don't really know why (unless we look at logs). Consider adding fprintf(stderr, "...") or improving error logging so it's easier to debug when things go wrong.
The PR solves the main issue well and keeps the code clean and readable. Just a few small additions and tests would help make it more complete and production-ready.

File: main.c [hunk 1]
```diff
@@ -16,4 +16,5 @@ int main(int argc, char **argv) {
 
 
 	sc_apply_seccomp_filter(&prog_allow);
+	fclose(file);
```
Comment (Dav3-K3rn) [inline main.c:19]: This file closing should also be added to error paths to prevent file descriptor leaks.

File: seccomp.c [hunk 2]
```diff
@@ -2,22 +2,18 @@
 #include <limits.h>
 #include <stdarg.h>
 #include <stdint.h>
-#include <stdio.h>
```
Comment (Dav3-K3rn) [inline seccomp.c:5]: This header can be removed, but the die() function still uses va_list and varargs.

File: seccomp.c [hunk 3]
```diff
@@ -2,22 +2,18 @@
...
 #include <stdlib.h>
 #include <unistd.h>
 
 #include <sys/prctl.h>
 #include <sys/stat.h>
 #include <sys/syscall.h>
 
-#include <linux/filter.h>
```
Comment (Dav3-K3rn) [inline seccomp.c:13]: Was <linux/filter.h> unused in the original code? Seccomp typically needs BPF filter definitions. Please verify this doesn't break any BPF-related functionality.

File: seccomp.c [hunk 4]
```diff
@@ -2,22 +2,18 @@
...
 #include <linux/prctl.h>
 #include <linux/seccomp.h>
 
 #include "seccomp.h"
 
-#define MAX_BPF_SIZE 32*1024
```
Comment (Dav3-K3rn) [inline seccomp.c:19]: You're removing the MAX_BPF_SIZE constant from this file?

File: seccomp.c [hunk 5]
```diff
@@ -2,22 +2,18 @@
...
-
 void die(const char *msg, ...)
 {
 	va_list ap;
```
Comment (Dav3-K3rn) [inline seccomp.c:19]: A validation function needs to be added. Then this function needs to be called after reading the header file.
static int validate_seccomp_header(const struct sc_seccomp_file_header *hdr) {
       if (hdr->header[0] != 'S' || hdr->header[1] != 'C') {
           die("Invalid seccomp header: %02x%02x", 
               hdr->header[0], hdr->header[1]);
       }
       if (hdr->version != 1) {  // or whatever valid version(s)
           die("Unsupported seccomp version: %u", hdr->version);
       }
       return 0;
   }
